lib/babe/babe.go

```diff
@@ -167,10 +172,13 @@ func (b *Session) stop() {
 	b.lock.Lock()
 	defer b.lock.Unlock()
 
-	if !b.closed {
+	if atomic.LoadUint32(&b.closed) == uint32(1) {
+		canUnlock := atomic.CompareAndSwapUint32(&b.closed, 1, 0)
+		if !canUnlock {
+			panic("[babe] Error when trying to change Service status from started to stopped.")
```

Comment:
return error or log

Reply:
also, this is changing `b.closed` from 1 to 0, which doesn't really make sense since we are closing it. maybe change the variable name to `stopped` ?

Reply:
I updated this to started, please re-check.

Reply:
Also updated the return.